urwid/tests/test_event_loops.py

```diff
@@ -11,38 +13,48 @@
 class EventLoopTestMixin:
     def test_event_loop(self):
         rd, wr = os.pipe()
-        evl = self.evl
+        evl: urwid.EventLoop = self.evl
         out = []
+
         def step1():
             out.append("writing")
             os.write(wr, b"hi")
+
         def step2():
             out.append(os.read(rd, 2).decode('ascii'))
             raise urwid.ExitMainLoop
-        handle = evl.alarm(0, step1)
-        handle = evl.watch_file(rd, step2)
+
+        _handle = evl.alarm(0, step1)
+        _handle = evl.watch_file(rd, step2)
+
         evl.run()
+        time.sleep(0.1)
```

Comment:
good to add explanation of why this is needed as a comment here

Reply:
Extra validation shown ineffectiveness of sleep